src/scripts/destinyTrackerApi/itemListBuilder.js

```diff
@@ -0,0 +1,74 @@
+import _ from 'underscore';
+import { itemTransformer } from './itemTransformer.js';
+
+class itemListBuilder {
+  constructor() {
+    this._itemTransformer = new itemTransformer();
+  }
+
+  getNewItems(allItems, scoreMaintainer) {
+    var self = this;
+    var allDtrItems = _.map(allItems, function(item) { return self._itemTransformer.translateToDtrWeapon(item); });
+    var allKnownDtrItems = scoreMaintainer.getItemStores();
+
+    var unmatched = _.filter(allDtrItems, function(dtrItem) {
+      var matchingItem = _.findWhere(allKnownDtrItems, { referenceId: String(dtrItem.referenceId), roll: dtrItem.roll });
+      return (matchingItem === null);
+    });
+
+    return unmatched;
+  }
+
+  getAllItems(stores) {
+    var allItems = [];
+
+    stores.forEach(function(store) {
+      allItems = allItems.concat(store.items);
+    });
+
+    return allItems;
+  }
+
+  getWeapons(stores, scoreMaintainer) {
```

Comment:
These methods are generically named and undocumented, so it's unclear to me what the difference between, for example, `getWeapons` and `getWeaponsList` would be.

Naming "private" methods with a leading underscore would help clear up the API too.